Empire/Empire.cpp

```diff
@@ -930,18 +1001,16 @@ void ProductionQueue::Update() {
         return;
     }
 
-    if (m_queue.empty()) {
-        //DebugLogger() << "ProductionQueue::Update aborting early due to an empty queue";
-        m_projects_in_progress = 0;
-        m_object_group_allocated_pp.clear();
-
-        ProductionQueueChangedSignal(); // need this so BuildingsPanel updates properly after removing last building
-        return;                         // nothing to do for an empty queue
-    }
-
     ScopedTimer update_timer("ProductionQueue::Update");
 
-    std::map<std::set<int>, float> available_pp = AvailablePP(empire->GetResourcePool(RE_INDUSTRY));
+    auto industry_resource_pool = empire->GetResourcePool(RE_INDUSTRY);
+    std::map<std::set<int>, float> available_pp = AvailablePP(industry_resource_pool);
+    int stockpile_location_id = industry_resource_pool->StockpileObjectID();
+    float available_stockpile = industry_resource_pool->Stockpile();
+    float stockpile_transfer_efficiency = empire->GetMeter("METER_IMPERIAL_PP_TRANSFER_EFFICIENCY")->Current();
```

Comment:
should probably rename this meter to clarify what it means... METER_STOCKPILE_INPUT_EFFIC or similar. don't think the imperial meter needs "IMPERIAL" in its name, though...

Reply:
Oh didnt see this comment. The name stems from the idea that the PP are the important thing and not the stockpile. IMPERIAL_PP would be to distinguish from "local" supply group PP.
But i think stockpile is the term that sticked and there is only the imperial one, so it it not necessary to point that out in the name. So I'll follow your suggestion in my branch.